source/How-To-Guides/Launch-file-different-formats.rst

```diff
@@ -10,12 +10,321 @@ Using Python, XML, and YAML for ROS 2 Launch Files
    :local:
 
 ROS 2 launch files can be written in Python, XML, and YAML.
-This guide shows how to use these different formats to accomplish the same task, as well as has some discussion on when to use each format.
+This guide shows how to use these different formats to accomplish the same task. It also discusses when to use each format.
 
-Launch file examples
---------------------
+Examples
+--------
+
+Taking arguments
+^^^^^^^^^^^^^^^^
```

Comment:
Other examples I would like to see (probably in their own tutorials so they are googleable):

* passing command line arguments to nodes
* passing arguments to a sub-launch file
* setting parameters for nodes by yaml file, by running the node with ros2 run, via the launch file, via command line argument to the launch file, and via the command line after the node is launched

Reply:
Sorry I meant "passing command line arguments to nodes in a launch file" (e.g. the argument goes through the launch file to the node).  For the last bullet point, I guess you're right - I diverged from launch files a little.  I think showing how to pass a yaml config file to a node inside the launch file and how to set parameters via the launch file and via the launch file plus command line argument is all launch related though.  Maybe a "How Parameters and Launch Files Interact" document.

As someone mentioned elsewhere, not everything I'm suggesting is probably suitable for an intro document but I think I'd like to see these all exist maybe with links from the intro document.  I do think discoverability is a big problem and I'm not sure what to do about that except to keep linking everything to everything...

Reply:
@jbarry-bdai it's there! See the `Configuring nodes` example. It includes a launch file, passing it an argument that is then passed as a command line argument to a node.

Reply:
Gah sorry don't know how I missed that!